File: InterviewQuestions/Javascript Interview Programming Questions/task3.js
// Write a JavaScript function to check if a given string is a palindrome


// Method 1: With builtin functions

const Palindrome1 = word => {
    let string_to_array = word.split('')
    let reverse_of_array = string_to_array.reverse()
    let new_string = reverse_of_array.join('')

    if (word === new_string) return "This is a palindrome"
    else return "This is not a palindrome"
}

let result1 = Palindrome1('tiger')

console.log(result1, ", result1")


// Method 2: Manual

const Palindrome2 = word => {
    let new_string = ''

    for (let x of word) new_string = x + new_string

    if (word === new_string) return "This is a palindrome"
    else return "This is not a palindrome"
}

let result2 = Palindrome2("reviver")

console.log(result2, ",result2")

File: InterviewQuestions/Python Interview Programming Questions/task3.py
# Write a Python function to check if a given string is a palindrome


# Method 1: Using BuiltIns

def Palindrome1(word):

    return "This is a palindrome" if word == word[::-1] else "This is not a palindrome"

result1 = Palindrome1("lion")

print(result1)


# Method 2: Manual


def Palindrome2(word):

    new_string = ''

    for x in word: new_string = x + new_string

    return "This is a palindrome" if word == new_string else "This is not a palindrome"

result2 = Palindrome2("reviver")

print(result2)

